modules/order/src/Form/UseMultiProfileConfirmForm.php

```diff
@@ -0,0 +1,390 @@
+<?php
+
+namespace Drupal\commerce_order\Form;
+
+use Drupal\Core\Config\FileStorage;
+use Drupal\Core\Config\StorageInterface;
+use Drupal\Core\Entity\EntityFieldManager;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Form\ConfirmFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Routing\CurrentRouteMatch;
+
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * A form to confirm the use of multiple profiles for shipping and billing.
+ */
+class UseMultiProfileConfirmForm extends ConfirmFormBase {
+
+  /**
+   * The current order type.
+   *
+   * @var \Drupal\commerce_order\Entity\OrderTypeInterface
+   */
+  protected $entity;
+
+  /**
+   * The config storage.
+   *
+   * @var \Drupal\Core\Config\StorageInterface
+   */
+  protected $configStorage;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The entity field manager.
+   *
+   * @var \Drupal\Core\Entity\EntityFieldManager
+   */
+  protected $entityFieldManager;
+
+  /**
+   * Constructs a ContentEntityForm object.
+   *
+   * @param \Drupal\Core\Routing\CurrentRouteMatch $current_route_match
+   *   The current route match.
+   * @param \Drupal\Core\Config\StorageInterface $config_storage
+   *   The config storage.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Drupal\Core\Entity\EntityFieldManager $entity_field_manager
+   *   The configurable field manager service.
+   */
+  public function __construct(
+    CurrentRouteMatch $current_route_match,
+    StorageInterface $config_storage,
+    EntityTypeManagerInterface $entity_type_manager,
+    EntityFieldManager $entity_field_manager
+  ) {
+    $this->entity = $current_route_match->getParameter('commerce_order_type');
+    $this->configStorage = $config_storage;
+    $this->entityTypeManager = $entity_type_manager;
+    $this->entityFieldManager = $entity_field_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('current_route_match'),
+      $container->get('config.storage'),
+      $container->get('entity_type.manager'),
+      $container->get('entity_field.manager'),
+      $container->get('commerce.configurable_field_manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getDescription() {
+    // Check if there are existing orders on this order type.
+    $description = $this->getExistingOrderCountDescription();
+
+    $description .= '<strong>'
+      . $this->t('This action cannot be undone. You cannot switch back to
+       using the single profile again.')
+      . '</strong>';
```

Comment:
It would be good to add a warning message like the following as well:

"If you choose to proceed, profiles for existing orders will be migrated to use separate profile types. That can take some time and it might cause errors if the users try to view their existing orders during the process; if you are running this on a production site please make sure that you are in maintenance mode".